datastore/miitopia/types/mii_tube_mii_info.go

```diff
@@ -0,0 +1,121 @@
+// Package types implements all the types used by the DataStoreMiitopia protocol
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/PretendoNetwork/nex-go/v2/types"
+	datastore_types "github.com/PretendoNetwork/nex-protocols-go/v2/datastore/types"
+)
+
+// MiiTubeMiiInfo is a type within the DataStoreMiitopia protocol
+type MiiTubeMiiInfo struct {
+	types.Structure
+	MetaInfo    *datastore_types.DataStoreMetaInfo
+	Category    *types.PrimitiveU8
+	RankingType *types.PrimitiveU8
+}
+
+// WriteTo writes the MiiTubeMiiInfo to the given writable
+func (mtmi *MiiTubeMiiInfo) WriteTo(writable types.Writable) {
+	contentWritable := writable.CopyNew()
+
+	mtmi.MetaInfo.WriteTo(writable)
```

Comment:
There was a bug on how structures were written. These should look like: `mtmi.MetaInfo.WriteTo(contentWritable)` instead. Can you fix it on every structure?

Reply:
whoops! thats certainly not good